beamer/check/commands.py

```diff
@@ -0,0 +1,256 @@
+        proof_source=proof_source,
+        proof_target=proof_target,
+        request_id=request_id,
+        fill_id=fill_id,
+        txhash=receipt.transactionHash,  # type: ignore
+        finalization_timestamp=None,
+    )
+
+
+@click.group()
+def check() -> None:
+    pass
+
+
+@check.command("initiate-l1-invalidations")
+@click.option(
+    "--keystore-file",
+    type=click.Path(exists=True, file_okay=True, dir_okay=False, path_type=Path),
+    required=True,
+    metavar="PATH",
+    help="Path to the keystore file.",
+)
+@click.password_option(
+    "--password",
+    type=str,
+    default="",
+    prompt=False,
+    help="The password needed to unlock the keystore file.",
+)
+@click.option(
+    "--rpc-file",
+    type=click.Path(exists=True, file_okay=True, dir_okay=False, path_type=Path),
+    required=True,
+    help="Path to the RPC config file.",
+)
+@click.option(
+    "--artifacts-dir",
+    type=click.Path(exists=True, file_okay=False, dir_okay=True, path_type=Path),
+    required=True,
+    help="The directory containing deployment artifacts.",
+)
+@click.option(
+    "--abi-dir",
+    type=click.Path(exists=True, file_okay=False, dir_okay=True, path_type=Path),
+    required=True,
+    help="Path to the directory with contract ABIs.",
+)
+@click.option(
+    "--output",
+    type=click.Path(file_okay=True, dir_okay=False, path_type=Path),
+    required=True,
+    help="Path to store the invalidation info at, which can be later used for verification.",
+)
+@click.option(
+    "--count",
+    type=click.IntRange(1),
+    default=1,
+    show_default=True,
+    help="Number of invalidations to create, per (PROOF_SOURCE, PROOF_TARGET) pair.",
+)
+@click.argument("proof-source", type=ChainIdParam())
+@click.argument("proof-target", type=ChainIdParam(), nargs=-1, required=True)
+def initiate_l1_invalidations(
+    keystore_file: Path,
+    password: str,
+    rpc_file: Path,
+    artifacts_dir: Path,
+    abi_dir: Path,
+    output: Path,
+    count: int,
+    proof_source: ChainId,
+    proof_target: tuple[ChainId],
+) -> None:
+    """Initiate L1 invalidations from chain PROOF_SOURCE to each PROOF_TARGET.
+    PROOF_SOURCE and PROOF_TARGET are chain IDs."""
+    beamer.util.setup_logging(log_level="DEBUG", log_json=False)
+
+    account = beamer.util.account_from_keyfile(keystore_file, password)
+    log.info("Loaded keystore file", address=account.address)
+
+    abi_manager = ABIManager(abi_dir)
+    deployment = beamer.artifacts.load(artifacts_dir, proof_source)
+
+    rpc_info = beamer.util.load_rpc_info(rpc_file)
+    url = rpc_info[proof_source]
+    w3 = make_web3(url, account)
+    assert w3.eth.chain_id == proof_source
+    log.info("Connected to RPC", url=url)
+
+    fill_manager = obtain_contract(w3, abi_manager, deployment, "FillManager")
+    request_manager = obtain_contract(w3, abi_manager, deployment, "RequestManager")
+    finality_period = request_manager.functions.chains(proof_source).call()[0]
```

Comment:
nitpick: you can probably move this call down to line 212, right before the for loop where the timestamps will be set. 

This would safe us the call if no invalidations would have been made.